runtime/commonMain/src/kotlinx/serialization/json/internal/JsonReader.kt

```diff
@@ -175,32 +175,58 @@ internal class JsonReader(private val source: String) {
         length += addLen
     }
 
+    fun nextJsonKey() {
+        val source = source
+        var currentPosition = currentPosition
+        val maxLen = source.length
+        // TODO this one can be optimized when we oll our benchmarks
+        while (true) {
+            if (currentPosition >= maxLen) {
+               fail("Expected next key name, but had trailing comma instead", currentPosition)
```

Comment:
Why not `Unexpected EOF`? Json can't end on trailing comma without `}` or `]` anyway